src/subprocess-posix.cc

```diff
@@ -165,15 +167,17 @@ ExitStatus Subprocess::Finish() {
 #endif
 
   if (WIFEXITED(status)) {
-    int exit = WEXITSTATUS(status);
-    if (exit == 0)
-      return ExitSuccess;
-  } else if (WIFSIGNALED(status)) {
+    // propagate the status transparently
+    return static_cast<ExitStatus>(WEXITSTATUS(status));
+  }
+  if (WIFSIGNALED(status)) {
+    // Overwrite interrupts to exit code 2
     if (WTERMSIG(status) == SIGINT || WTERMSIG(status) == SIGTERM
         || WTERMSIG(status) == SIGHUP)
       return ExitInterrupted;
   }
-  return ExitFailure;
+  // At this point, we exit with any other signal+128
+  return static_cast<ExitStatus>(status | 0x80);
```

Comment:
why not `+ 128`?

Reply:
It's hard to say; I've seen it in some sources during the first research.